xsuite/src/cli/newWalletCmd.ts

```diff
@@ -72,14 +76,21 @@ const action = async ({
     data = newWallet.toJSON();
   }
   if (data === undefined) {
-    const mnemonic = Mnemonic.generate().toString();
-    data = UserWallet.fromMnemonic({ mnemonic, password }).toJSON();
+    try {
+      const mnemonic = createMnemonic(shard);
```

Comment:
The `createMnemonic` abstraction is not needed and if its code is inlined, the error thrown could be replaced by a `return logError(...)` and the try/catch would not be needed anymore.